### DISCUSSION TITLE
Annotations' clusters

### DISCUSSION THREAD

**Discussion by vaggarath** [google/model-viewer#3538] [Q&A] [answered]
I'm scraching the back of my head since this morning about this "option" but i can't seem to find a work around to get it. 
I have my annotations from a JSON, (very simple construction it's the annotations' coordinates and their text), i make and i generate programmaticaly the buttons once the model and the json are fully loaded. It works like a charm, (so far, so good). What i'd like to achieve is the possibility to make clusters (like those on maps like gmap by exemple) of button while comparing if the annotations are more or less close from each others.
And since i'm trully not a pain in the ... I don't want to touch the current JSON's construction model (the annotations i have today will evolve when i'll no longer be there and the guys who will work with it won't be able to do nesting in a json).
Would anyone have tips to help me (well, if you already made something like this and willing to share i won't be shy!) ? ^^

Also i don't know if it's good idea but couldn't it be a function implemented in the lib ?

**Comment by elalish** [comment, marked as answer]
> Also i don't know if it's good idea but couldn't it be a function implemented in the lib ?

Anything with that much UX subtlety I'd prefer to keep outside. Probably the simplest way would be to have a fixed hierarchy of cluster nodes, add them all as hotspots, but change their visibility based on `getCameraOrbit`; maybe just the radius, or perhaps some more complex math.

**Reply by vaggarath** [reply]
I went with your proposition. Should've thought about it before ><

Thx a lot !
